### PR TITLE
feat: include Trade tx info

### PR DESCRIPTION
**Pull request by zzmp**
Include more usable classes in the swap/wrap transaction info to facilitate more featureful integrations.

- Refactors wrap logic to re-use the existing `TransactionType`
- Includes fully instantiated `Trade` and `CurrencyAmount` in transaction info

### CODE REVIEW

File: src/components/Swap/SwapButton/index.tsx [hunk 1]
```diff
@@ -173,11 +174,10 @@ export default memo(function SwapButton({ disabled }: SwapButtonProps) {
   }, [approvalAction, disableSwap, onReviewSwapClick, isPending, onWrap, trade.state, wrapType])
   const Label = useCallback(() => {
     switch (wrapType) {
```
Comment (vm) [inline src/components/Swap/SwapButton/index.tsx:176]: this variable should be `transactionType` now, plus wherever else in this file

Comment (vm) [reply src/components/Swap/SwapButton/index.tsx:176]: or leave as WrapType?

File: src/components/Swap/Status/StatusDialog.tsx [hunk 2]
```diff
@@ -20,7 +26,7 @@ const TransactionRow = styled(Row)`
   flex-direction: row-reverse;
 `
 
-type PendingTransaction = Transaction<SwapTransactionInfo | WrapTransactionInfo>
+type PendingTransaction = Transaction<SwapTransactionInfo | WrapTransactionInfo | UnwrapTransactionInfo>
```
Comment (vm) [inline src/components/Swap/Status/StatusDialog.tsx:29]: why can't this be `Transaction<TransactionInfo>`?

**Review by vm — APPROVED** [review]
Some of the naming of `WrapType` vs `TransactionType` is confusing. I might consider leaving only `TransactionType` and then having a function called `isWrap` which returns true of Wrap | Unwrap.

File: src/components/Swap/Status/StatusDialog.tsx (hunk 2 above)
Comment (zzmp) [reply src/components/Swap/Status/StatusDialog.tsx:29]: NB: That's equivalent to just `Transaction` (as `TransactionInfo` defaults).

File: src/hooks/swap/useWrapCallback.tsx [hunk 3]
```diff
@@ -5,15 +5,13 @@ import { useWETHContract } from 'hooks/useContract'
...
 import tryParseCurrencyAmount from 'utils/tryParseCurrencyAmount'
 
 import useCurrencyBalance from '../useCurrencyBalance'
 
-export enum WrapType {
-  NONE,
-  WRAP,
-  UNWRAP,
-}
+export type WrapType = TransactionType.WRAP | TransactionType.UNWRAP | undefined
```
[the remark this answers is not in the document]
Comment (zzmp) [reply src/hooks/swap/useWrapCallback.tsx:13]: only directly below it - I'll just inline the usage.

**Comment by zzmp** [comment]
> Some of the naming of `WrapType` vs `TransactionType` is confusing. I might consider leaving only `TransactionType` and then having a function called `isWrap` which returns true of Wrap | Unwrap.

I added `isWrap` to the `useWrapCallback` return value, and made a new helper hook, `useIsWrap`, to cover this use case and clarify the code. Hope that resolves this 🤞

**Pull request merged**